src/spinner/src/Spinner.js

```diff
@@ -38,84 +38,69 @@ const innerClass = color =>
     fill: 'transparent'
   }).toString()
 
-class Spinner extends PureComponent {
-  static propTypes = {
-    /**
-     * Composes the Box component as the base.
-     */
-    ...Box.propTypes,
+function Spinner({ delay, size, ...props }) {
+  const theme = useTheme()
+  const [isVisible, setIsVisible] = useState(delay === 0)
+  const [delayTimer, setDelayTimer] = useState(null)
 
-    /**
-     * Delay after which spinner should be visible.
-     */
-    delay: PropTypes.number,
-
-    /**
-     * The size of the spinner.
-     */
-    size: PropTypes.number.isRequired,
-
-    /**
-     * Theme provided by ThemeProvider.
-     */
-    theme: PropTypes.object.isRequired
-  }
-
-  static defaultProps = {
-    size: 40,
-    delay: 0
-  }
-
-  constructor({ delay }) {
-    super()
-
-    this.state = {
-      isVisible: delay === 0
+  useEffect(() => {
+    if (delay > 0) {
+      const newDelayTimer = setTimeout(() => {
+        setIsVisible(true)
+      }, delay)
+      setDelayTimer(newDelayTimer)
```

Comment:
I dont think we even need to use state for this – we know the `id` and can return that as part of the effect cleanup return function.